plugins/main/public/components/common/globalBreadcrumb/platformBreadcrumb.tsx

```diff
@@ -6,34 +6,38 @@ export const setBreadcrumbs = (breadcrumbs, router) => {
   if (breadcrumbs === '' || breadcrumbs === undefined) {
     return;
   }
-  const breadcrumbsCustom = breadcrumbs?.map(breadcrumb =>
-    breadcrumb.agent
-      ? {
-          className:
-            'euiLink euiLink--subdued osdBreadcrumbs wz-vertical-align-middle',
-          onClick: ev => {
-            ev.stopPropagation();
-            if (getWzCurrentAppID() === endpointSummary.id) {
-              NavigationService.getInstance().navigate(
-                `/agents?tab=welcome&agent=${breadcrumb.agent.id}`,
-              );
-            } else {
-              NavigationService.getInstance().navigateToApp(
-                endpointSummary.id,
-                {
-                  path: `#/agents?tab=welcome&agent=${breadcrumb.agent.id}`,
-                },
-              );
-            }
+  const breadcrumbsCustom = breadcrumbs
+    ?.map(breadcrumb =>
+      breadcrumb?.agent?.id
+        ? {
+            className:
+              'euiLink euiLink--subdued osdBreadcrumbs wz-vertical-align-middle',
+            onClick: ev => {
+              ev.stopPropagation();
+              if (getWzCurrentAppID() === endpointSummary.id) {
+                NavigationService.getInstance().navigate(
+                  `/agents?tab=welcome&agent=${breadcrumb.agent.id}`,
+                );
+              } else {
+                NavigationService.getInstance().navigateToApp(
+                  endpointSummary.id,
+                  {
+                    path: `#/agents?tab=welcome&agent=${breadcrumb.agent.id}`,
+                  },
+                );
+              }
+            },
+            truncate: true,
+            text: breadcrumb.agent.name,
+          }
+        : typeof breadcrumb.agent !== 'undefined'
+        ? null
```

Comment:
**suggestion**: explaining the use case could help in the future to understand the reason of it.